### ISSUE TITLE
Optimize imported VectorDrawables before loading them into Shape Shifter

### ISSUE THREAD

**Issue by KennyGoers** [alexjlockwood/ShapeShifter#302]
When Import the attached Vector Image it adds an extra absolute movement onto it, no idea why
[map_pin_motion.txt](https://github.com/alexjlockwood/ShapeShifter/files/2236666/map_pin_motion.txt)

Imported the path for the "main" part is "M 53 61 M 42 61 C 42 58.084 43.16 55.284 45.222 53.222 C 47.284 51.16 50.084 50 53 50 C 55.916 50 58.716 51.16 60.778 53.222 C 62.84 55.284 64 58.084 64 61 C 64 63.916 62.84 66.716 60.778 68.778 C 58.716 70.84 55.916 72 53 72 C 50.084 72 47.284 70.84 45.222 68.778 C 43.16 66.716 42 63.916 42 61 Z" (note the extra at the start)

**Comment by alexjlockwood** [comment]
So the `<path>` named `main` in the vector drawable looks like this:

```
  <path
      android:name="main"
      android:pathData="M 53,61 m -11,0 a 11,11 0,1 1,22 0 a 11,11 0,1 1,-22 0 Z"
      android:strokeWidth="2"
      android:fillColor="#20BFDC"
      android:strokeColor="#FFFFFF"
      android:fillType="evenOdd"/>
</vector>
```

And that is resulting in the following imported path:

```
// main
M 53 61 M 42 61 C 42 58.084 43.16 55.284 45.222 53.222 C 47.284 51.16 50.084 50 53 50 C 55.916 50 58.716 51.16 60.778 53.222 C 62.84 55.284 64 58.084 64 61 C 64 63.916 62.84 66.716 60.778 68.778 C 58.716 70.84 55.916 72 53 72 C 50.084 72 47.284 70.84 45.222 68.778 C 43.16 66.716 42 63.916 42 61 Z
```

The only difference here really is that the elliptical arcs are being converted to bezier curves. Both of the paths have the same number of subpaths here (the original begins with `M m` and the imported one begins with `M M`. So I don't think it is correct to say that an extra M is being added.

Did you mean to suggest that the extra M should be optimized away?

**Comment by KennyGoers** [comment]
You are right, I ran into it when trying to sync up a few paths for animation and it was difficult to do so because of the 'added' point, and in tracking it down I noticed it was unnecessary. so optimizing it away would probably be the correct way of handling it...

**Comment by alexjlockwood** [comment]
Cool, I renamed the bug.

Shape Shifter actually does optimize these things away for imported SVGs... I just haven't written it for imported VectorDrawables yet...
